File: WPRK-iOS/WPRK-iOS/Domain/Data/PodcastDTO/PodcastsDTO.swift
//
//  PodcastsDTO.swift
//  WPRK-iOS
//
//  Created by Mwai Banda on 6/25/22.
//

import Foundation

// MARK: - PodcastsMetaData
struct PodcastsDTO: Codable {
    let collection: [PodcastDTO]
    let metaData: PodcastsMetaData
    enum CodingKeys: String, CodingKey {
        case collection = "data"
        case metaData = "meta"
    }
}

File: WPRKSDK/src/commonMain/kotlin/com/mwaibanda/wprksdk/data/podcastDTO/PodcastsDTO.kt
package com.mwaibanda.wprksdk.data.podcastDTO

import com.mwaibanda.wprksdk.data.metadataDTO.MetadataDTO
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class PodcastsDTO(
    @SerialName("data")
    val collection: List<PodcastDTO>,
    val meta: MetadataDTO
)

